src/api_server/src/parsed_request.rs

```diff
@@ -294,15 +294,45 @@ pub(crate) mod tests {
         }
     }
 
+    fn http_response(body: &str, status_code: i32) -> String {
+        let header = format!(
+            "HTTP/1.1 {} \r\n\
+             Server: Firecracker API\r\n\
+             Connection: keep-alive\r\n",
+             status_code
+        );
+        
+        let content = format!(
+            "Content-Type: application/json\r\n\
+             Content-Length: {}\r\n\r\n{}",
+             body.len(),
+             body,
+        );
+
+        if status_code == 204 { // No Content
+            return format!("{}{}", header, "\r\n");
+        }
+        format!("{}{}", header, content)
+    }
+
+    fn http_request(request_type: &str, endpoint: &str, body: &str) -> String {
+        format!("{} {} HTTP/1.1\r\n\
+            Content-Type: application/json\r\n\
+            Content-Length: {}\r\n\r\n\
+            {}",
+            request_type,
+            endpoint,
+            body.len(),
+            body 
+        )
+    }
+
     #[test]
     fn test_missing_slash() {
         let (mut sender, receiver) = UnixStream::pair().unwrap();
         let mut connection = HttpConnection::new(receiver);
-        sender
-            .write_all(
-                b"GET none HTTP/1.1\r\n\
-                Content-Type: text/plain\r\n\
```

Comment:
Here, "text/plain" is used as the Content-Type, while the **http_request** function will default it to 'application/json' for every case. For this to be correct and retain the same behaviour, I believe we should also be able to set the content type.

Reply:
if `Content-Type: application/json` also works in all tests, I say we use that as the default and stop using `Content-Type: text/plain`